### PR TITLE
GSM8k with assertions notebook

### PR DESCRIPTION
**Pull request by Shangyint**

### CODE REVIEW

File: examples/math/gsm8k/gsm8k_assertions.ipynb [hunk 1]
```diff
@@ -0,0 +1,653 @@
...
+        "%autoreload 2\n",
+        "\n",
+        "import sys\n",
+        "import os\n",
+        "import regex as re\n",
+        "import json\n",
+        "\n",
+        "import pkg_resources # Install the package if it's not installed\n",
+        "if not \"dspy-ai\" in {pkg.key for pkg in pkg_resources.working_set}:\n",
+        "    !pip install git+https://github.com/stanfordnlp/dspy.git@accenture-course\n",
```
Comment (arnavsinghvi11) [inline examples/math/gsm8k/gsm8k_assertions.ipynb:115]: does this notebook need to install from this branch? I believe the latest changes here have been merged to main

File: examples/math/gsm8k/gsm8k_assertions.ipynb [hunk 2]
```diff
@@ -0,0 +1,653 @@
...
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "w1rZSbu2pQY3"
+      },
+      "outputs": [],
+      "source": [
+        "prediction = basic_math_solver(question=\"What is 1+1+1?\")\n",
```
Comment (arnavsinghvi11) [inline examples/math/gsm8k/gsm8k_assertions.ipynb:289]: the cache seems to break for me here on just this query. the rest works however!

**Review by arnavsinghvi11 — CHANGES_REQUESTED** [review]

**Comment by arnavsinghvi11** [comment]
Thanks @Shangyint ! left a few comments regarding the notebook. should be good to merge after that.

**Comment by Shangyint** [comment]
@arnavsinghvi11 Thanks! Fixed both issues.

**Comment by arnavsinghvi11** [comment]
Thanks!

**Pull request merged**